### PR TITLE
Add analysis module

### PR DESCRIPTION
**Pull request by roshankern**
@gwaybio This is ready for review!

### CODE REVIEW

**Comment by roshankern** [comment]
Just realized it might be useful to show well or plate for UMAP colors, which could indicate/disprove any plate/image effects on features. I will implement this and let you know when done.

File: 2.analyze_data/2.analyze_data.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Step 0: Convert notebook to script
+jupyter nbconvert --to=script 2.analyze_training_data.ipynb
+
+# Step 1: Execute jupyter notebook
+jupyter nbconvert --to=html \
+        --ExecutePreprocessor.kernel_name=python3 \
+        --ExecutePreprocessor.timeout=10000000 \
+        --execute 2.analyze_training_data.ipynb
```
Comment (gwaybio) [inline 2.analyze_data/2.analyze_data.sh]:
```suggestion
        --execute 2.analyze_training_data.ipynb

```

**Review by gwaybio — APPROVED** [review]
LGTM! A couple general comments:

1. I like that you saved the UMAP coordinates, will you use these to generate publication-ready figures later? (I noticed that you didn't save the actual UMAP images)
2. Does `format_training_data.ipynb` need a number prefix? (i.e. does it need to be executed in a specific order?)

**Comment by roshankern** [comment]
1. I was just being thorough saving the UMAP coordinates (in case we wanted to analyze/visualize these differently). The publication-ready figures are accessible through the executed jupyter notebook html file. Should I also save the images in `results/`?
2. `format_training_data.ipynb` does not need to be executed in a specific order so I will remove the number prefix.

**Comment by gwaybio** [comment]
> Should I also save the images in results/?

Probably. It wouldn't hurt.

**Pull request merged**